scripts/verify-pr.py

```
@@ -0,0 +1,608 @@
+    Raises:
+        ValueError: If PR identifier contains invalid characters or patterns
+    """
+    # Allow:
+    # - Numeric PR IDs (e.g., "7", "123")
+    # - Branch names (alphanumeric, dots, hyphens, underscores, slashes)
+    # - Comparison refs (e.g., "main...feature", "base..head")
+    # - Hash prefixes with # (e.g., "#7")
+    valid_pattern = r"^[#]?[A-Za-z0-9._/-]+(?:\.\.\.[A-Za-z0-9._/-]+)?$"
+
+    if not pr_identifier or not pr_identifier.strip():
+        raise ValueError("PR identifier cannot be empty")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)